test-app/tests/unit/models/sprite-tree-test.ts

```diff
@@ -408,4 +414,110 @@ module('Unit | Models | SpriteTree', function (hooks) {
+          parentContext.element,
+          'ContextAsSprite',
+          childContext.element
+        );
+        subject.addSpriteModifier(contextSpriteModifier);
+
+        leafSpriteModifier = new MockSpriteModifier(
+          childContext.element,
+          'LeafSprite'
+        );
+        subject.addSpriteModifier(leafSpriteModifier);
+      });
+      test('nodes are correctly added and typed', function (assert) {
+        let childNode = subject.lookupNodeByElement(
+          childContext.element
+        ) as SpriteTreeNode;
+
+        assert.ok(childNode, 'child node exists');
+        assert.strictEqual(childNode.isContext, true, 'node is context');
+        assert.strictEqual(childNode.isSprite, true, 'node is sprite');
+        assert.strictEqual(
+          childNode.children.size,
+          1,
+          'context node has a single child'
+        );
+
+        let leafSprites = [...childNode.children];
+        assert.strictEqual(
+          leafSprites.length,
+          1,
+          'childContext node has a single child'
+        );
+
+        let leafSprite = leafSprites[0] as SpriteTreeNode;
+        assert.ok(leafSprite.isSprite);
+        assert.notOk(leafSprite.isContext);
+        assert.strictEqual(leafSprite.spriteModel, leafSpriteModifier);
+      });
+      test('getting descendants', function (assert) {
+        let parentContextDescendants = subject.descendantsOf(parentContext);
+        assert.deepEqual(
+          parentContextDescendants,
+          [childContext, contextSpriteModifier, leafSpriteModifier],
+          'includes childContext as AnimationContext instance, childContext as SpriteModifier instance, and leafSprite as SpriteModifier instance'
+        );
+
+        let childContextDescendants = subject.descendantsOf(childContext);
+        assert.deepEqual(
+          childContextDescendants,
+          [leafSpriteModifier],
+          'includes leafSprite as SpriteModifier instance'
+        );
+      });
+      test('getting a context run list', function (assert) {
+        assert.deepEqual(
+          subject.getContextRunList(new Set([parentContext])),
+          [parentContext],
+          'run list for the parent context just includes the parent context'
+        );
+        assert.deepEqual(
+          subject.getContextRunList(new Set([parentContext, childContext])),
+          [childContext, parentContext],
+          'when both parent and child are specified both are returned, with child first'
+        );
+        assert.deepEqual(
+          subject.getContextRunList(new Set([childContext])),
+          [childContext, parentContext],
+          'when a child is specified, the run list includes the child then the parent'
+        );
+        let runList = subject.getContextRunList(new Set([childContext]));
+        assert.ok(
+          runList[0] === childContext,
+          'when two children are specified, the parent is included once, after the children'
```

Comment:
Should the message here be changed? Since there is only one child context now.

Reply:
Yep 😬